### DISCUSSION TITLE
What performance can be expected?

### DISCUSSION THREAD

**Discussion by muellerma** [jxlsteam/jxls#203] [Q&A] [answered]
Hi,

I have a template with 37 columns and row count will be about 30k.
Now I'm testing a bit in my local dev env and seeing horrible performance.
I've measured the duration of the applyAt method.
1k rows take about 1 minute.
2k rows take about 7 minutes.
5k rows take about 63 minutes.

Am I doing something wrong?
My template is a simple one without any formulars, sums or other stuff. Just a tx:foreach to write the rows.
I've adapted the [simpleSxssf() Demo](https://github.com/jxlsteam/jxls/blob/6b5417ad3e018661caede45f33a266d7ccc67e07/jxls-poi/src/test/java/org/jxls/examples/stress/SxssfDemo.java#L45).

Thanks

**Comment by SoltauFintel** [comment, marked as answer]
One minute for 1,000 rows is too much.

Please create a new workspace using the jxls repository. Then run SxssfDemo.java as a unit test. For me, the test case takes 5.2 seconds and processes 30,000 rows. Then change the test case step by step. Execute the test case again and again in between. This way you can determine which change is causing the performance error. Please only use MS Excel.

There are no performance measurements in the jxls project that I can share. jxls is used in a number of programs every day. With larger amounts of data, the use of streaming is important so that POI does not have to keep so many XML objects in memory.

**Comment by muellerma** [comment]
Hi,

Thanks for answering. In the meantime I got it working much more performant in a range you've mentioned.
It looked like my xlsx template was the problem. I recreated it using an example in the documentation and now it runs in a few seconds. Don't know what the problem was exactly, but more important to me is that it's running and I can use it for my use-case as planned.
If you're interested in analyzing the issue I could provide the template.

**Reply by SoltauFintel** [reply]
No, thanks. ;-)
